lib/src/main/kotlin/com/swmansion/starknet/data/types/Event.kt

```diff
@@ -64,16 +64,16 @@ data class EmittedEvent @JvmOverloads constructor(
 @Serializable
 data class GetEventsPayload @JvmOverloads constructor(
     @SerialName("from_block")
-    val fromBlockId: BlockId,
+    val fromBlockId: BlockId? = null,
 
     @SerialName("to_block")
-    val toBlockId: BlockId,
+    val toBlockId: BlockId? = null,
 
     @SerialName("address")
-    val address: Felt,
+    val address: Felt? = null,
 
     @SerialName("keys")
-    val keys: List<List<Felt>>,
+    val keys: List<List<Felt>>? = null,
```

Comment:
When reviewing and checking if fields is required or not, please see [`EVENT_FILTER`](https://github.com/starkware-libs/starknet-specs/blob/6d88b7399f56260ece3821c71f9ce53ec55f830b/api/starknet_api_openrpc.json#L992) and [`RESULT_PAGE_REQUEST`](https://github.com/starkware-libs/starknet-specs/blob/6d88b7399f56260ece3821c71f9ce53ec55f830b/api/starknet_api_openrpc.json#L898).